sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -135,38 +135,40 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
         }
     }
 
-    private Flux<ByteBuffer> validateLength(final HttpRequest request) {
+    static Flux<ByteBuffer> validateLength(final HttpRequest request) {
         final Flux<ByteBuffer> bbFlux = request.getBody();
         if (bbFlux == null) {
             return Flux.empty();
         }
 
         long expectedLength = Long.parseLong(request.getHeaders().getValue("Content-Length"));
-        final long[] currentTotalLength = new long[1];
 
-        return Flux.concat(bbFlux, Flux.just(VALIDATION_BUFFER)).handle((buffer, sink) -> {
-            if (buffer == null) {
-                return;
-            }
+        return Flux.defer(() -> {
+            final long[] currentTotalLength = new long[1];
+            return Flux.concat(bbFlux, Flux.just(VALIDATION_BUFFER)).handle((buffer, sink) -> {
```

Comment:
just curious, is this issue happens due to one of these operators eagerly subscribing? or something else?

Reply:
I see the state variables were getting shared across subscriptions on request content.